packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -74,6 +74,11 @@ class Observatory {
     _getEventController(data['streamId']).add(event);
   }
 
+  void populateIsolateInfo() {
+    // Calling this has the side effect of populating the isolate information.
+    waitFirstIsolate;
```

Comment:
It's a getter, defined immediately below. I wrapped it in another method in order to make it a bit more semantic about what it's doing :)

We populate the list of isolates with isolate created events (and isolate terminated events). This call will also fire off a call to get the VMRef object, which contains the full list of isolates known at the time. We had been missing the main isolate if we connected the debugger after it had been created.